### PR TITLE
chore: making init more interactive

### LINKED ISSUE #2195: feat: tailcall init should be interactive
Tailcall init needs to be more interactive, here is the proposed design.

```
> tailcall init my-new-project
  ? File Format:
    - GraphQL
    - JSON
    - YML
  [INFO] File write: ./my-new-project/.tailcallrc.schema.json
  [INFO] File write: ./my-new-project/.tailcallrc.graphql
  [INFO] File write: ./my-new-project/.graphqlrc.yml
  [INFO] File write: ./my-new-project/main.graphql  
```

**Technical Requirements**

- Logs will be generated automatically if the runtime from CLI is used.
- Err if the user doesn't set a path to the init command.
- Create the directory recursively if it doesn't exit
- Create the files in the newly generated directory.
- By default create a file called `main` with the extension selected above.
- Create a simple Hello World config inside of the main.
  ```graphql
  schema {
    query: Query
  }
  type Query {
    greet: String @const(body: "Hello World!")
  }
  ```

### PR DESCRIPTION
**Pull request by thesapan**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #2195
/claim 2195

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/cli/tc.rs [hunk 1]
```diff
@@ -95,6 +98,21 @@ pub async fn run() -> Result<()> {
 }
 
 pub async fn init(folder_path: &str) -> Result<()> {
+    let project_name = Text::new("Project Name:").prompt().unwrap_or_else(|err| {
+        eprintln!("Error: {}", err);
+        std::process::exit(1);
```
Comment (ssddOnTop) [inline src/cli/tc.rs:103]: can not exit,  throw an error instead

File: src/cli/tc.rs [hunk 2]
```diff
@@ -95,6 +98,21 @@ pub async fn run() -> Result<()> {
...
+    });
+
+    let chosen_format = Select::new(
+        "Choose a file format:",
+        vec![Source::GraphQL, Source::Json, Source::Yml],
+    )
+    .prompt()
+    .unwrap_or_else(|err| {
+        eprintln!("Error: {}", err);
```
Comment (ssddOnTop) [inline src/cli/tc.rs:112]: use `tracing::error!` instead of `eprintln`

**Review by tusharmath — COMMENTED** [review]
Please fix comments.

**Comment by tusharmath** [comment]
Closing this PR because it doesn't do exactly what the issue says. We need to create the files in the user-specified folder

**Pull request closed without merging**